torchtune/modules/tokenizers/__init__.py

```diff
@@ -4,15 +4,31 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-from ._sentencepiece import SentencePieceBaseTokenizer
-from ._tiktoken import TikTokenBaseTokenizer
-from ._utils import (
+# NOTE: This file is maintained for backward compatibility purposes.
+# The imports below point to the new location in `torchtune.modules.transforms.tokenizers`.
+# The import paths will be removed in v0.7. Please update your code to use the new path
+# (torchtune.modules.transforms.tokenizers) to avoid breaking changes in future releases.
+
+
+import warnings
+
+from torchtune.modules.transforms.tokenizers import (
     BaseTokenizer,
     ModelTokenizer,
     parse_hf_tokenizer_json,
+    SentencePieceBaseTokenizer,
+    TikTokenBaseTokenizer,
     tokenize_messages_no_special_tokens,
 )
 
+warnings.warn(
+    "The import path 'torchtune.modules.tokenizers' is deprecated and will be removed in v0.7. "
+    "Please update your imports to 'torchtune.modules.transforms.tokenizers'.",
+    DeprecationWarning,
+    stacklevel=2,
+)
+
+
 __all__ = [
```

Comment:
Technically we can probably delete `__all__` here since its purpose is to declare public APIs for the module and we don't want people importing from here (it won't break anything to do this since we still import). But no strong preference, we can also just remove when we inevitably delete this file.